src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -656,9 +664,22 @@ void DoorLockServer::GetCredentialStatusCommandHandler(
     SuccessOrExit(err = commandObj->PrepareCommand(path));
     VerifyOrExit((writer = commandObj->GetCommandDataIBTLVWriter()) != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
     SuccessOrExit(err = writer->Put(TLV::ContextTag(to_underlying(ResponseFields::kCredentialExists)), credentialExists));
-    if (0 != userIndexWithCredential)
+    if (credentialExists)
     {
-        SuccessOrExit(err = writer->Put(TLV::ContextTag(to_underlying(ResponseFields::kUserIndex)), userIndexWithCredential));
+        if (0 != userIndexWithCredential)
+        {
+            SuccessOrExit(err = writer->Put(TLV::ContextTag(to_underlying(ResponseFields::kUserIndex)), userIndexWithCredential));
```

Comment:
Is there a a reason we're not just defining a `Commands::GetCredentialStatusResponse::Type`, filling it in, and then handing it to the CommandHandler, instead of doing this manual TLV stuff?

Reply:
No particular reason, actually. When I was implementing that I didn't know that was possible. I should definitely fix although I'm more inclined to do that in a separate PR

Reply:
Created new issue to track this: #18396